src/librustc_resolve/lib.rs

```diff
@@ -1969,10 +1996,27 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                 Err(())
             }
         } else {
+
+            // find possible candidates
+            let trait_name = trait_path.segments.last().unwrap().identifier.name;
+            let candidates: Vec<_> =
+                self.lookup_suggestions(
+                    trait_name,
+                    LookupTarget::AllCrates,
+                    Namespace::TypeNS,
+                    move |def: Def|
+                        match def {
+                            Def::Trait(_) => true,
+                            _             => false
+                        }
+                );
+
             resolve_error(self,
                           trait_path.span,
-                          ResolutionError::UndeclaredTraitName(&path_names_to_string(trait_path,
-                                                                                      path_depth)));
+                          ResolutionError::UndeclaredTraitName(
+                            &path_names_to_string(trait_path, path_depth),
+                            candidates
```

Comment:
Nit: if you're going to move the `)` to the next line, you ought to have a trailing comma for better diffs in the future :)